src/check-label/check-label.ts

```diff
@@ -0,0 +1,66 @@
+// check-label/check-label.ts
+
+import process from 'node:process';
+import path from 'node:path';
+import assert from 'node:assert';
+import { readFile } from 'node:fs/promises';
+import { debug } from 'debug';
+import { setFailed } from '@actions/core';
+import semver from 'semver';
+
+import { getFileFromMain, getLabelsOnPR } from '../github-api';
+
+const log = debug('check-label');
+
+interface PackageJSON {
+  name: string;
+  version: string;
+  files: string[];
+}
+
+async function getLocalPackageJsonVersion(fileName: string): Promise<string> {
+  const packageJSONPath = path.join(process.cwd(), fileName);
+  const readPackageJson = await readFile(packageJSONPath, 'utf8');
+  const packageJson = JSON.parse(readPackageJson) as PackageJSON;
+
+  return packageJson.version;
+}
+export default async function (): Promise<void> {
+  log('Action start');
+
+  const labelsPullRequest = await getLabelsOnPR();
+  if (labelsPullRequest.length > 1) {
+    throw new Error('PR has more than one label');
+  }
+  const label = labelsPullRequest[0]?.toLowerCase();
+  assert(label, 'Unable to get label from PR');
+
+  const branchPackageJsonVersion = await getLocalPackageJsonVersion('package.json');
+  const mainPackageJsonVersionRaw = await getFileFromMain('package.json');
+
+  if (!mainPackageJsonVersionRaw) {
+    throw new Error('Unable to get package.json from main branch');
+  }
+  const mainPackageJsonVersion = JSON.parse(mainPackageJsonVersionRaw) as PackageJSON;
+
+  const packageVersionDiff = semver.diff(branchPackageJsonVersion, mainPackageJsonVersion.version);
```

Comment:
I think this check should be more precise.  e.g. if main has a version of 1.3.1, and the new PR has a version of 2.7.1, it would be ok if the label was "major".  I think instead we should apply the same logic as our current AP to the main version and make sure it exactly matches whats in the PR.  e.g. 1.3.1 -> 2.0.0 if the label is "major".

Reply:
Makes sense, I've implemented and fixed the tests. Doing tests with github tokens are trick when using actions for the tests..